### PR TITLE
Feature/update animals table

### PR DESCRIPTION
**Pull request by Zafron047**
### Dear Reviewer,

This pull request includes the completion of various tasks related to the animals table and transaction handling. The following tasks have been accomplished:

- Added a new column "species" of type string to the animals table.
- Inserted animal data with details like name, birthdate, weight, neutered status, and escape attempts.
- Executed changes within transactions to modify and verify species column updates.
- Updated animals' species based on specific conditions within transactions, committing and verifying changes.
- Calculated statistics and answered queries regarding animal data.

Screenshots of transaction outcomes and query results have been provided for documentation.

# **Tasks completed and files updated:**
- Updated schema.sql to include "species" column.
- Updated data.sql with new animal insertions.
- Added transactions and queries to queries.sql.
- Added screenshots of transaction outcomes and query results.

##  **Screenshots of Transactions:**

- [x] **Transaction# 1 Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.**

- [x] ** Transaction# 2 Inside a transaction:**
  - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
  - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Verify that changes were made.
  - Commit the transaction.
  - Verify that changes persist after commit.

- [x] **Transaction# 3 Inside a transaction delete all records in the animals table**
  - Inside a transaction delete all records in the animals table
Delete PNG

  - Rollback the transaction and after the rollback verify if all records in the animals table still exists. 
Delete_undo

- [x] **Transaction# 4 Inside a transaction:**
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.

  - Rollback to the savepoint

  - Update all animals' weights that are negative to be their weight multiplied by -1.

  - Commit transaction
4 4_COMMIT

##  **Screenshots of Queries:**
 - [x] **Query 1 Write queries to answer the following questions:**
  - How many animals are there?

  - [x] **Query 2 How many animals have never tried to escape?**

  - [x] **Query 3 What is the average weight of animals?**

  - [x] **Query 4 Who escapes the most, neutered or not neutered animals?**

  - [x] **Query 5 What is the minimum and maximum weight of each type of animal?**

  - [x] **Query 6 What is the average number of escape attempts per animal type of those born between 1990 and 2000?**

## **Fixed required change by Reviewer**
 - [x] **You should add a statement to verify if all records in the animals' table still exist after rolling back.**
Req_changes

Please kindly review and merge this pull request to incorporate the changes made to the project.

Thank you!

@Zafron047

### CODE REVIEW

File: querries.sql [hunk 1]
```diff
@@ -28,4 +30,117 @@ WHERE name <> 'Gabumon';
...
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+
+-- Inside a transaction
+BEGIN;
+
+-- delete all records
+DELETE FROM animals;
+
+-- then roll back the transaction
+ROLLBACK;
```
Comment (AbelG101) [inline querries.sql:93]: - [ ] Ouch! you're so close, but as part of the project requirements you should add a statement to verify if all records in the animals' table still exist after rolling back but you seem to be missing that in your queries file here.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @Zafron047 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Species column is properly added ✔
- Correct queries for each question ✔
- Proper transaction management ✔
- Very well-organized SQL statements✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great screenshot summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: querries.sql (hunk 1 above)
Comment (Zafron047) [reply querries.sql:93]: ### **Fixed**
you should add a statement to verify if all records in the animals' table still exist after rolling back.

```shell
ROLLBACK;

-- After the rollback verify if all records in the animals table still exists.
SELECT COUNT(*) FROM animals;

``` 

Req_changes

**Comment by Zafron047** [comment]
## **Required change by Reviewer**

- [x] **you should add a statement to verify if all records in the animals' table still exist after rolling back.**

```shell
ROLLBACK;

-- After the rollback verify if all records in the animals table still exists.
SELECT COUNT(*) FROM animals;
``` 

Req_changes

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Zafron047, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
@AbelG101 Thanks for the thorough inspection and your kind review. 😺

**Comment by Zafron047** [comment]
@mukhammadrizooff Thanks a lot for your kind review and approval. 😄

**Pull request merged**